Sources/StreamChat/Deprecations.swift

```diff
@@ -213,3 +213,14 @@ public typealias NotificationInviteAccepted = NotificationInviteAcceptedEvent
 
 @available(*, deprecated, renamed: "UnknownChannelEvent")
 public typealias UnknownEvent = UnknownChannelEvent
+
+public extension CurrentChatUserController {
+    @available(
+        *,
+        deprecated,
+        message: "use addDevice(_pushDevice:) instead. This deprecated function doesn't correctly support multiple push providers."
+    )
+    func addDevice(token: Data, pushProvider: PushProvider = .apn, completion: ((Error?) -> Void)? = nil) {
```

Comment:
We agreed to keep the deprecations in the same file as they were before

Reply:
I thought this was a v5+ approach agreement. If we do it now, the deprecations will be all over the place. I think we should try to make it consistent for now

Reply:
Hmmm, seems like we've been having different approaches on that then 😅 . I saw other PRs where deprecations were kept. We even created a ticket in JIRA to move the ones in this file to its original file (https://stream-io.atlassian.net/browse/CIS-1552)

Reply:
I've moved it 👍

Reply:
Moving deprecation back to files is yak shaving, IMHO. We can start following the new approach when the v5 is released. For now, the consistency matters more

Reply:
I agree with @evsaev, but now I've changed it already 😄 I think we should follow the new approach only after that ticket is complete.